Attachment support

hdoddema opened the pull request:
Hello,

I have implemented sending attachments and inline attachments according to [the Mailjet API v3 documentation](https://dev.mailjet.com/guides/#sending-with-attached-files-v3).

`attach`, `attachContent`, `embed` and `embedContent` should be available now.

hdoddema committed:
init support for attachments

hdoddema committed:
debug

hdoddema committed:
debug

hdoddema committed:
apparently we're API v3

hdoddema committed:
remove debug stuff

hdoddema committed:
TO/BCC/CC init

hdoddema committed:
TO/BCC/CC init

hdoddema committed:
TO/BCC/CC init

Julian-B90 commented on Mailer.php:
```diff
@@ -131,30 +131,55 @@ public function getResponse()
         ];
 
+        if ($cc) {
+            $body['Cc'] = join(', ',$cc);
```
missing space `join(', ', $cc);`

Julian-B90 commented on Mailer.php:
```diff
@@ -131,30 +131,55 @@ public function getResponse()
 
+        if ($cc) {
+            $body['Cc'] = join(', ',$cc);
+        }
```
Add Space

Julian-B90 commented on Message.php:
```diff
@@ -113,6 +117,9 @@ public function getCc() {
      * @inheritdoc
      */
     public function setCc($cc) {
+        if (!is_array($cc)){
```
Please add space `if (!is_array($cc)) {`

Julian-B90 commented on Message.php:
```diff
@@ -128,6 +135,9 @@ public function getBcc() {
      * @inheritdoc
      */
     public function setBcc($bcc) {
+        if (!is_array($bcc)){
```
Please add space `if (!is_array($bcc)) {`

Julian-B90 reviewed (commented):
Can you maybe add an example for Attachments and InlinedAttachments

hdoddema committed:
formatting, fix for inline attachments

hdoddema committed:
fix for inline attachments

hdoddema commented:
Oh yes, and in the meantime I have also changed Recipients to separate fields for To, CC and BCC. It seems to work ok, but I'm not sure if using Recipients had any advantages over doing it like this. 

But this way it also supports setBcc and setCc as you would expect.

I have fixed a few bugs with the inline attachments just now.

Here is some sample code.

```php
// Mail with attachment from string via Message::attachContent()
Yii::$app->mailer->compose('view-name')
->setSubject('Mail with attachment from content')
->attachContent("This is the attachment content",['fileName'=>'attachment.txt', 'contentType'=>'text/plain'])
->setTo('info@example.com')
->send();

// Mail with attachment from file via Message::attach()
$filePath = ... // a file path here;
Yii::$app->mailer->compose('view-name')
->setSubject('Mail with attachment from content')
->attach($filePath)
->setTo('info@example.com')
->send();
```
Embedding allows you to use the image directly in the HTML content. It does require a different approach because you need a reference to the message object to get the image links.


```php
$message = Yii::$app->mailer->compose('view-name')
    ->setSubject('Mail with inline attachment')
    ->setTo('info@example.com')
$imageLink = $message->embed($filePath);
$message->setHtmlBody('Sample embedded image <br> <img src="'.$imageLink.'">');
$message->send();
```

Julian-B90 commented:
Thank you very much for your work 👍
